14/README.md

```diff
@@ -134,28 +136,35 @@ For this BEP, we analyze a round-robin connection strategy, similar to [Round-ro
 In this scenario, every request is forwarded to a different node, distributing the load to all the nodes, and allowing recovering from connection errors by forwarding the failed request to another healthy node.
 
 This strategy needs to keep track of the following values:
-- `E`, the list of BigchainDB endpoints passed at initialization time (we use the Python notation `len(L)` to get the length of the list).
-- `i`, the index of the BigchainDB node to use for a request (`0 ≤ i < len(L)`).
-- `T`, a list of timestamps. It has the same length of `L`. On initialization, all elements contain `0`. It is used to track the availability of the nodes. We will see in a second how it is used.
-- `DELAY`, in case a node is unreachable, how much milliseconds we should wait. This is used in combination with `T`.
+
+- `E`, the list of BigchainDB endpoints passed at initialization time.
+- `i`, the index of the BigchainDB node to use for a request (`0 ≤ i < len(E)`).
+- `timeout`, user-specified timeout, defaults to 20 seconds. Users may choose to have no timeout.
+- `T`, a list of timestamps, expressed as the number of seconds elapsed since Unix Epoch. It has the same length as `E`. On initialization, all elements contain the minimal possible timestamp `0`. It is used to track the availability of the nodes.
+- `DELAY`, in case a node is unreachable, how much seonds we must pass before the request can be repeated. This is used in combination with `T`.
+- `retry_count`, how many times the node was retried unsuccessfully.
 
 We also need to know the current time of the system, for this we use the variable `current_time_ms`.
 
 When a driver needs to forward a request to the network, the following algorithm (or equivalent) SHALL be implemented:
 
-1. if `current_time_ms > T[i]`:
-    1. forward the request to the node `N[i]`
-    1. if the request failed, we need to avoid hitting the unhealthy node for some time `T[i] = current_time_ms + DELAY`
-    1. move the index to the next node `i = (i + 1) mod len(N)`
-    1. return the result or the error
-1. if `current_time_ms ≤ T[i]`:
-    1. move the index to the next node `i = (i + 1) mod len(N)`
-    1. return error
+1. pick the node `i` with the minimal `T[i]`, that is the earliest available node
+1. wait until `current_time_ms > T[i]` or timeout is expired
+1. if timeout is expired, return
+1. if timeout is not expired, make the request
+1. if timeout is expired while the request is in progress, return
+1. in case of a successful response, set `retry_count` to 0, set `T` to `datetime.min`, return
```

Comment:
Should `datetime.min` be `0`?

Reply:
Yes! Forgot to update it here, good catch.